src/klein/resource.py

```diff
@@ -280,3 +280,39 @@ def write_response(r):
         d.addCallback(write_response).addErrback(log.err, _why="Unhandled Error writing response")
 
         return server.NOT_DONE_YET
+
+
+
+class KleinHTTPRequest(server.Request):
+
+    def getArg(self, key):
+        """
+        Get a single arg value.
+
+        @raises KeyError: If key doesn't exist
+        @raises ValueError: If there is more than 1 value
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        value = self.args[key]
+        if len(value) != 1:
+            raise ValueError('Too many values for: {0}'.format(key))
+        return value[0]
+
+    def getArgs(self, key):
+        """
+        Get the list of values for a key.
+
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        return self.args.get(key, [])
+
+
+
+class KleinSite(server.Site):
+    def buildProtocol(self, addr):
+        channel = http.HTTPFactory.buildProtocol(self, addr)
+        channel.requestFactory = KleinHTTPRequest
```

Comment:
Have you considered using the [`requestFactory` argument to `Site.__init__`](https://twistedmatrix.com/documents/17.1.0/api/twisted.web.server.Site.html#__init__) rather than having a custom subclass?